hera_qm/utils.py

```diff
@@ -722,10 +724,13 @@ def apply_yaml_flags(uv, a_priori_flag_yaml, lat_lon_alt_degrees=None, telescope
             antnum = ant[0]
         if issubclass(uv.__class__, UVData):
             blt_selection = np.logical_or(uv.ant_1_array == antnum, uv.ant_2_array == antnum)
-            uv.flag_array[blt_selection, :, :, pol_selection] = True
+            if np.count_nonzero(blt_selection) > 0:
+                for bltind in np.where(blt_selection)[0]:
+                    uv.flag_array[bltind, :, :, pol_selection] = True
         elif issubclass(uv.__class__, UVCal):
             ant_selection = uv.ant_array == antnum
-            uv.flag_array[ant_selection, :, :, :, pol_selection] = True
-
+            if np.count_nonzero(ant_selection) > 0:
```

Comment:
I prefer `np.any(blt_selection)` but that's stylistic.

Reply:
I agree.